src/fsharp/PrettyNaming.fs

```diff
@@ -713,4 +713,24 @@ module public FSharp.Compiler.PrettyNaming
                 | _ -> Some (defaultArgName, actualArgValue))
         mangleProvidedTypeName (nm, nonDefaultArgs)
 
-    let outArgCompilerGeneratedName = "outArg"
\ No newline at end of file
+    let outArgCompilerGeneratedName = "outArg"
+
+    [<AutoOpen>]
+    module GeneratedFieldNames =
```

Comment:
Can a comment be  added here explaining the reasoning in the PR? I just predict a future someone (perhaps me!) reading this in a few years and getting horribly confused.

Reply:
Perhaps also stating that 10 names is likely to be enough, since most DUs won't have more than 10 cases

Reply:
I've added xml docs.